module/documents/actor/actor.mjs

```diff
@@ -1488,15 +1488,27 @@ export default class Actor5e extends Actor {
       if ( Number.isNumeric(abilityValue) ) parts.push(String(abilityValue / 100));
     }
 
-    // Create the d20 roll
-    const formula = parts.join(" + ");
-    return new CONFIG.Dice.D20Roll(formula, data, {
-      flavor: flavor ?? game.i18n.localize("DND5E.Initiative"),
-      advantageMode: advantageMode,
+    options = foundry.utils.mergeObject({
+      flavor: options.flavor ?? game.i18n.localize("DND5E.Initiative"),
       halflingLucky: flags.halflingLucky ?? false,
       critical: null,
       fumble: null
-    });
+    }, options);
+
+    /**
+     * A hook event that fires before initiative is rolled for an Actor.
+     * @function dnd5e.preRollInitiative
+     * @memberof hookEvents
+     * @param {Actor5e} actor                The Actor that is rolling initiative.
+     * @param {object} data                  The roll data.
+     * @param {string[]} parts               The roll parts.
+     * @param {D20RollConfiguration} config  Configuration data for the pending roll.
+     */
+    Hooks.callAll("dnd5e.preRollInitiative", this, data, parts, options);
```

Comment:
A main reason I *did not* add these hooks here is I really don't like the mess of having a pre-hook here in `getInitiativeRoll` and a post-hook in an overridden `rollInitiative()` method. This is - to me - really ugly and I would vote to remove it and either not offer hooks here or offer a different framing of hooks.